src/protocolluaapi.cpp

```diff
@@ -0,0 +1,232 @@
+	auto appName = msg.getString();
+
+	switch (recvbyte) {
+		// Ping & Pong to ensure that server or api is available and is responding
+		case 100: {
+			setResponse(true);
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::sendCallbackMessage, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+				appName, "pong was successful")));
+			return;
+		}
+		// This will execute whatever data it gets through the string directly as lua code via "dostring" function
+		case 101: {
+			// We use fileName, to let the lua interface know how this file is named, incase it throws an error so we get a name to it.
+			auto fileName = msg.getString();
+			auto data = msg.getString();
+			auto returnvalue = g_scripts->executeString(data, fileName);
+			if (!returnvalue.empty()) {
+				g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::sendErrorMessage, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+					appName, returnvalue)));
+				return;
+			}
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::sendCallbackMessage, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+				appName, "successfully executed " + fileName)));
+			return;
+		}
+		// This parses packets received as strings and then creates a lua table representating the data given from the http headers & values
+		case 102: {
+			auto eventName = msg.getString();
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::parseHttpRequest, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+				appName, eventName, msg)));
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::sendCallbackMessage, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+				appName, "http request received")));
+			return;
+		}
+		default:
+			g_dispatcher.addTask(createTask(std::bind(&ProtocolLuaApi::sendErrorMessage, std::static_pointer_cast<ProtocolLuaApi>(shared_from_this()),
+				 appName, "recvbyte: " + std::to_string(recvbyte) + " does not exist!")));
+	}
+}
+
+void ProtocolLuaApi::parseHttpRequest(std::string& appName, std::string& name, NetworkMessage& msg)
+{
+	std::unordered_map<std::string, std::string> dataMap;
+	// now we get a byte which tells us how much http headers we have to parse
+	auto headers = msg.get<uint16_t>();
+	for (uint16_t i = 0; i < headers; i++) {
+		// we get the header string
+		auto headerString = msg.getString();
+		// now we get the data as string
+		auto dataString = msg.getString();
+		// add everything to map
+		dataMap.emplace(headerString, dataString);
+	}
+
+	/* We are searching if we can even find a registered GlobalEvent with the name
+	*  This way we could push app names onward from the API to only execute the GlobalEvent which belongs to this app
+	*/
+	auto http = g_globalEvents->getEventMap(GLOBALEVENT_HTTPREQUEST);
+	if (http.find(name) != http.end()) {
+		g_globalEvents->getEventMap(GLOBALEVENT_HTTPREQUEST).at(name).executeHttpRequest(name, dataMap);
```

Comment:
using at but not capturing the exception anyway, so why using at?

Reply:
actually I just noticed that find holds the pointer already, that way I don't even need to use `at` in this case